bindings/java/java/src/main/java/org/ethereum/evmc/Host.java

```diff
@@ -146,7 +146,8 @@ static synchronized int addContext(HostContext context) {
   }
 
   static void removeContext(int index) {
-    contextList.remove(index);
+    HostContext context = contextList.remove(index);
+    context.close();
```

Comment:
Why not mark `HostContext` as `AutoCloseable` and then this is a given?

Reply:
`AutoCloseable` allows you to write syntactic sugar in java with the `try-with-resources` approach:
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Doesn't really apply here. Also that `close` method of `AutoCloseable` throws an Exception in its signature, so you'd have to deal with a try/catch.

Reply:
Actually after #548 this is not needed.